sdks/python/apache_beam/transforms/core.py

```diff
@@ -2475,7 +2475,7 @@ def typed(transform):
 
       def inject_default(_, combined):
         if combined:
-          assert len(combined) == 1
+          assert len(combined) >= 1
```

Comment:
In this case we've just changed the assertion to allow for "combined" longer than one. This doesn't throw an error but we do lose some info when we only return combined[0] one line later, though I had some trouble understanding what the significance of that was.
If that lost info doesn't matter, or if there's a better way to deal with it, I'm wondering if we need the assertion at all since the pipeline would just fail on the next line anyways.

Reply:
the intent of the `_inject_default` function is to add a default element in the combined result if the combined result is empty and a default value can be well-defined as `combine_fn.apply([], *args, **kwargs)`. 

Users can disable default values if they use `.without_defaults()` value when they apply a combiner.  

The replacement of the empty output with default value happens at the last step of combining. At this stage, PCollection being combined should be reduced to 1 element, or be empty. Default value is not necessary when the result of combining is NOT empty - in this case we should take the reduced result. However, since this is evaluated after prior combining, we expect that the result of prior combining is  a single value. Therefore, we have an assertion that checks that prior combining has reduced the pcollection to 1 element. If somehow we have more than 1 element, then the result may not be fully combined, so we should understand how that happened.